### PR TITLE
Milk Quests, Fix Dye action, added leather_horse_armor

### PR DESCRIPTION
**Pull request by pvillaverde**
Hey there

I was setting a few common quests for jobs and found that Milk actions didn't work except on the Jobs that pays for milking(Farmer). Debugging I found this piece of code is the answer.

I'm not sure why is it there, because when checking quests or paying it already checks that out. So check if there's something I missed. As far as I tested, it now works, and doesn't pay nor anything if user hasn't got that action on its jobs.
```java


	boolean found = false;
	t: for (JobProgression prog : jPlayer.getJobProgression()) {
	    for (JobInfo info : jPlayer.getJobProgression(prog.getJob()).getJob().getJobInfo(ActionType.MILK)) {
		if (info.getActionType() == ActionType.MILK) {
		    found = true;
		    break t;
		}
	    }
	}

	if (!found) {
	    return;
	}
```
Later, when I was testing dye quests, I found that the order in which you put the dye and the leather item caused the action to get done or not. Therefore if you set the dye as the first item, it didn't work.  I also took the opportunity to add the LEATHER_HORSE_ARMOR as an item that can be dyed.

I Also found that while shear action was getting paid, shear quests didn't work. On the process quest, when it checks if the COLOR of the sheep is on the objectives, it's case-sensitive, so I modified the jobConfig docs to change that to uppercase. (I wasn't sure how to make that ignoring uppercase, would like to if it's possible, to avoid any future headache).

Commit:
```
- Fix: Milk quests not working if palyer doesn't have a job which pays for milking.
- Fix: Dye action not working if dye is the first item on crafting
- Feature: Added LEATHER_HORSE_ARMOR as item which can be dyed.
- Docs: Specified caps on Shear quest. Otherwise it doesn't work.
```

### CODE REVIEW

**Comment by montlikadani** [comment]
> I was setting a few common quests for jobs and found that Milk actions didn't work except on the Jobs that pays for milking(Farmer). Debugging I found this piece of code is the answer.
> 
> I'm not sure why is it there, because when checking quests or paying it already checks that out. So check if there's something I missed. As far as I tested, it now works, and doesn't pay nor anything if user hasn't got that action on its jobs.
> 
> ```java
> 	boolean found = false;
> 	t: for (JobProgression prog : jPlayer.getJobProgression()) {
> 	    for (JobInfo info : jPlayer.getJobProgression(prog.getJob()).getJob().getJobInfo(ActionType.MILK)) {
> 		if (info.getActionType() == ActionType.MILK) {
> 		    found = true;
> 		    break t;
> 		}
> 	    }
> 	}
> 
> 	if (!found) {
> 	    return;
> 	}
> ```

This implementation includes a fix where the player is not in a suitable job where the Milk action is given, but in another, so the player receives a reward as well as a delay message.



> I Also found that while shear action was getting paid, shear quests didn't work. On the process quest, when it checks if the COLOR of the sheep is on the objectives, it's case-sensitive, so I modified the jobConfig docs to change that to uppercase. (I wasn't sure how to make that ignoring uppercase, would like to if it's possible, to avoid any future headache).

It's not right. Recently, retrieving jobConfig action materials to memory has changed. So what the user has specified in the action are materials, either uppercase or lowercase. But lowercase is more noticeable than uppercase. To improve, the toLowerCase method in Quests is required to add materials to memory in small print.

**Comment by pvillaverde** [comment]
I've restored the implementation of that fix and added other check, thus checking if any job pays for milking or if any job has a quest with milking objective.As for the Shear actions. regular payment works fine and ignores case. The problem comes with quests, on this line Jobs:99:
```java
    if (!quest.getObjectives().containsKey(action.getType()) || !quest.getObjectives().get(action.getType()).containsKey(action.getNameWithSub()) && !quest.getObjectives().get(action.getType())
        .containsKey(action.getName()))
```
The method containsKey doesn't ignore case, neither the .get() on a hashMap.

The color comes on uppercase from `sheep.getColor().name()`on the onEntityShear Listener. I think the fix would be to either force the objectives to be uppercase or lowercase, and do the same with this colour. Otherwise, just inform that it must be specified on caps on the quests. 

Btw: I'm thinking of adding some spanish translations and maybe other feature related with placeholders. As for the translations, Do I make them and do other PR or is there other place to do them?

**Comment by montlikadani** [comment]
> Btw: I'm thinking of adding some spanish translations and maybe other feature related with placeholders. As for the translations, Do I make them and do other PR or is there other place to do them?

We have Crowdin for this.

File: src/main/resources/jobConfig.yml [hunk 1]
```diff
@@ -1742,7 +1742,7 @@ Jobs:
         Name: "Regular Joe"
         Objectives:
         - Breed;Sheep;10
-        - Shear;White;10
+        - Shear;WHITE;10
```
Comment (montlikadani) [inline src/main/resources/jobConfig.yml:1745]: Revert this. This needs some improvements in the code that the material names should be `lowerCase` even the user uses large letter.

Comment (pvillaverde) [reply src/main/resources/jobConfig.yml]: done!

**Comment by pvillaverde** [comment]
> 
> 
> > Btw: I'm thinking of adding some spanish translations and maybe other feature related with placeholders. As for the translations, Do I make them and do other PR or is there other place to do them?
> 
> We have Crowdin for this.

I'm checking it, its more advanced than the translations I had until now, but some of them aren't yet commited into the plugin itself, are they? Also I can't find the `TranslatableWords->Words_en.yml`, that one is a bit of a mess with many "Aire (Air)" translations.

**Comment by montlikadani** [comment]
We haven't added the TranslatableWords files to Crowdin yet, because there are still issues with older minecraft versions. So it will be added later. "Air" appears because the material you are using may not exist in the current version, so it shows Air or None.

**Comment by pvillaverde** [comment]
I see most of them are Materials from >= 1.13, but I'm running on 1.15.2, they should have their name. Instead users see they got paid by breaking air and things like that. Others that are on english also cause confussion, "Cooked Fish" as translation or "Cooked cod". And then there's also the parenthesis redundant with the english translation. 

I'll be keeping an eye for when it's available for translation, I'll be working on it meanwhile

**Pull request merged**